client/components/sync-handler.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect } from 'react';
+import { useSelector, shallowEqual } from 'react-redux'
+import isEqual from 'lodash/isEqual';
+import classnames from 'classnames';
+
+const selector = ({
+  project: version,
+  savedProject,
+  application: {
+    readonly,
+    editConditions
+  }
+}) => {
+  // only compare version data if version is in an editable state
+  const editable = !readonly || editConditions;
+  const isSyncing = editable && !isEqual(version, savedProject);
+  return { isSyncing };
+};
+
+const LoadingIndicator = () => {
```

Comment:
Can we sync the const name and the file name? :D

Reply:
I see what you did there.

They used to be synced, and then I changed one of them.